### PR TITLE
Remove PPStream protocol dissector

### PR DESCRIPTION
**Pull request by 41CEA55**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

P2P video player PPStream was discontinued shortly after the purchase of PPS.tv by Baidu (iQIYI) and the last available version online dates back to 2013.

Please note that the chinese video hosting service PPS.tv is still active, so I left the protocol id `NDPI_PROTOCOL_PPSTREAM` and the stuff in `ndpi_content_match.c.inc` untouched.

But it's probably better to rename this to `NDPI_PROTOCOL_PPS_TV` to avoid confusion.

https://www.techinasia.com/report-baidu-acquires-video-rival-pps

### CODE REVIEW

**Comment by IvanNardi** [comment]
> But it's probably better to rename this to `NDPI_PROTOCOL_PPS_TV` to avoid confusion.

Yes, please rename it (and add the new entry in `doc/protocols.rst`, please)

**Comment by IvanNardi** [comment]
I took a closer look at some past tests ans run some new ones:
1) iqiyi app version 4.1.0 (02/2022) still had some "ppstream" leftovers: the string "ppstream" was used in same HTTP user agent and some UDP flows matched the logic in `src/lib/protocol/ppstream.c`
2) iqiyi app version 6.3.0 (latest) seems to not use anything related to "ppstream" anymore but some small UDP flows are not classified (the heuristic in `src/lib/protocol/ppstream.c` is out of date,maybe?)
3) iqiyi.com seems quite different from pps.tv and ppstream.com

Two years is not a long time but I think we can update this stuff anyways. If someone is really interested in these protocols we can improve them later.

For the time being:
*) Remove `src/lib/protocol/ppstream.c`
*) Use `DPI_PROTOCOL_PPS_TV` for "pps.tv" and "ppstream.com"
*) Remove the two domain about "vidto"
*) Use `DPI_PROTOCOL_IQIYI` for the other 5 domains

What do you think?

**Comment by 41CEA55** [comment]
> I took a closer look at some past tests ans run some new ones:
> 
> 1. iqiyi app version 4.1.0 (02/2022) still had some "ppstream" leftovers: the string "ppstream" was used in same HTTP user agent and some UDP flows matched the logic in `src/lib/protocol/ppstream.c`
> 2. iqiyi app version 6.3.0 (latest) seems to not use anything related to "ppstream" anymore but some small UDP flows are not classified (the heuristic in `src/lib/protocol/ppstream.c` is out of date,maybe?)
> 3. iqiyi.com seems quite different from pps.tv and ppstream.com
> 
> Two years is not a long time but I think we can update this stuff anyways. If someone is really interested in these protocols we can improve them later.
> 
> For the time being: *) Remove `src/lib/protocol/ppstream.c` *) Use `DPI_PROTOCOL_PPS_TV` for "pps.tv" and "ppstream.com" *) Remove the two domain about "vidto" *) Use `DPI_PROTOCOL_IQIYI` for the other 5 domains
> 
> What do you think?

Well, you're right - iQIYI and PPS.tv apps create some UDP sessions when watching videos, so I'll try to investigate this issue better.

But there's one more thing: both apps access the same `iqiyi.com` domain, `pps.tv` is only mentioned in ServerNames. Does it make sense to keep `DPI_PROTOCOL_PPS_TV` just to deal with `pps.tv` and `ppstream.com`?

[iqiyi.log](https://github.com/ntop/nDPI/files/15064122/iqiyi.log)
[ppstv.log](https://github.com/ntop/nDPI/files/15064131/ppstv.log)

**Comment by 41CEA55** [comment]
> Well, you're right - iQIYI and PPS.tv apps create some UDP sessions when watching videos, so I'll try to investigate this issue better.

About the iQIYI app (from Google Play): yeah, some UDP packets have `PPStream` strings in the payload, but the offset to them is different... this is a job for `ndpi_memmem`! 😁 Also there are some short sessions that have matching 2-byte sequences in requests and responses, but they are unique for each session.

Apart from that, there is nothing special.

Both applications create sessions of two UDP packets containing JSON: some kind of authentication? The PPS.tv application creates some kind of TCP session, but I have no idea what it is. The main part of the traffic is QUIC and TLS.

I think it's worth it to take a closer look and make a separate dissector for iQIYI.

Here's some PCAP samples: [pcaps.zip](https://github.com/ntop/nDPI/files/15065877/pcaps.zip)

**Comment by IvanNardi** [comment]
> > Well, you're right - iQIYI and PPS.tv apps create some UDP sessions when watching videos, so I'll try to investigate this issue better.
> 
> About the iQIYI app (from Google Play): yeah, some UDP packets have `PPStream` strings in the payload, but the offset to them is different... this is a job for `ndpi_memmem`! 😁 Also there are some short sessions that have matching 2-byte sequences in requests and responses, but they are unique for each session.
> 
> Apart from that, there is nothing special.
> 
> Both applications create sessions of two UDP packets containing JSON: some kind of authentication? The PPS.tv application creates some TCP sessions, but I have no idea what it is. The main part of the traffic is QUIC and TLS.
> 
> I think it's worth it to take a closer look and make a separate dissector for iQIYI.
> 
> Here's some PCAP samples: [pcaps.zip](https://github.com/ntop/nDPI/files/15065877/pcaps.zip)
> 
> @IvanNardi what do you think?

I 'll look into that ASAP.
Meanwhile:
* could you report the versions of the apps that you used, please?
* What did you use for testing pps.tv? The app is no available on GoogleStore, isn't?

**Comment by 41CEA55** [comment]
> I 'll look into that ASAP. Meanwhile:
> 
> * could you report the versions of the apps that you used, please?
> * What did you use for testing pps.tv? The app is no available on GoogleStore, isn't?

I'm not sure about the iQIYI version, I installed the one that was on Google Play so it's probably the latest. Yea, PPS.tv app isn't available on Google Play, so I downloaded it here: https://pps.apponic.com/android/

**Comment by 41CEA55** [comment]
Here is the UDP traffic of the Windows version of the iQiyi app. Strange, but there are no `PPStream` strings here at all.

[iqiyi_windows_udp.zip](https://github.com/ntop/nDPI/files/15067527/iqiyi_windows_udp.zip)

**Comment by 41CEA55** [comment]
@IvanNardi  Unfortunately, I couldn't find any signatures in iQIYI UDP traffic, except for the mobile version - some sessions there contain the string `PPStream`. Everything else looks like random data.

Most of the traffic is TLS and QUIC.

**Comment by 41CEA55** [comment]
Here's some iQIYI UDP traffic samples if you're interested in.
[iqiy_udp.zip](https://github.com/ntop/nDPI/files/15078442/iqiy_udp.zip)

File: src/lib/ndpi_main.c [hunk 1]
```diff
@@ -2308,6 +2304,10 @@ static void ndpi_init_protocol_defaults(struct ndpi_detection_module_struct *ndp
 			  "BFCP", NDPI_PROTOCOL_CATEGORY_VIDEO,
 			  ndpi_build_default_ports(ports_a, 0, 0, 0, 0, 0) /* TCP */,
 			  ndpi_build_default_ports(ports_b, 0, 0, 0, 0, 0) /* UDP */);
+  ndpi_set_proto_defaults(ndpi_str, 0 /* encrypted */, 0 /* nw proto */, NDPI_PROTOCOL_FUN, NDPI_PROTOCOL_IQIYI,
+			  "iQIYI", NDPI_PROTOCOL_CATEGORY_VIDEO,
```
Comment (IvanNardi) [inline src/lib/ndpi_main.c]: `NDPI_PROTOCOL_CATEGORY_STREAMING`? Not a big difference, but I'd prefer to have the same category here and in the domains list

Comment (41CEA55) [reply src/lib/ndpi_main.c:2308]: > `NDPI_PROTOCOL_CATEGORY_STREAMING`? Not a big difference, but I'd prefer to have the same category here and in the domains list

Apparently I missed it, I'll fix it now.

**Pull request merged**